ViewMoreModal and ActionItemModal work for ActionItems

julianrkung opened the pull request:
implemented ViewMoreModal (generalized the CaseNoteCardModal) and the ActionItemModal now works for editing action items on the action item creation page. The final button layout might cause merge conflicts with Calvin's PR since his button layout will be different and conflict. AddToTemplates functionality doesn't work yet but can be added later/during hackathon (would be an easy addition since it's just an API call).

## Note:
I also had to tweak some of the styling for ActionItemCard because it wasn't rendering properly. Make sure that it renders nicely on your end when you pull before merging into master.

## To test:
Pull and edit action items that were both created from templates and from scratch and make sure nothing funky happens. When you edit an action item created from a template the checkmark goes away. If that's not what you want I can also change that.


### Screenshots
View modal
![image](https://user-images.githubusercontent.com/35501399/80873520-049e5d00-8c6e-11ea-9e67-4a361b69393c.png)

Edit modal
![image](https://user-images.githubusercontent.com/35501399/80873524-1122b580-8c6e-11ea-97a1-56a2e96a39af.png)

What the cards should look like
![image](https://user-images.githubusercontent.com/35501399/80873527-17189680-8c6e-11ea-8b30-a37f2d858ed6.png)



CC: @didvi

wilsonw926 commented:
Couple questions:

1. If you create from scratch and then add as a "common assignment" then you edit that assignment. should that edit also be reflected in common assignment tab? i.e when i edit "wqrwrwqer" (a new assignment created from scratch) in this example
![image](https://user-images.githubusercontent.com/22403821/80890989-03712e00-8c76-11ea-8ba6-51a1d4db2c4c.png)

2. going off of the above example, if i click 'edit' this "wqrwrwqer" assignment and edit it to "wqrwrwqer123123". then i check "add to common assignments", right now the new update assignment doesn't show up in the common assignment area.

wilsonw926 commented:
Also getting a `Uncaught TypeError: handleOpenModal is not a function` in actionitemlist

julianrkung commented:
Hey @wilsonw926 thanks for catching the TypeError. I changed one of the component's props and not the other so I just fixed that.

I should ask @didvi  for the expected behavior of editing. I think that only the LHS should be edited. Also the "Add to common assignments" doesn't work right now. I need to ask if it should always just create a new template or if the template should be edited if it came from an existing template.
